reactioncommerce/reaction
# Fixed archived variants being also cloned when top level variant is cloned

Resolves #2814 

I noticed what was causing this bug was as a result of the variants collection being fetched for the Product to be cloned didn't have a constraint to check whether the variant has been archived or not. So I added a constraint to check that ([find it here](https://github.com/reactioncommerce/reaction/blob/audax-fix-issue-2814/server/methods/catalog.js#L354)). I also added an `optionTitle` parameter with `- copy` appended to it to the `Object.assign` function to handle having "Untitled option" in the text fields after duplicating.

#### How to test
- Create a product
- Add a variant option
- Add a second variant option
- Archive and publish the second variant option
- Duplicate the top-level variant
- You would observe there are no archived variant options among the duplicated variants and option title for cloned variants is also present now.
